### ISSUE TITLE
FEM: Results of CalculiX simulations are displayed using 2 separate objects

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#10821]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Currently, there are two separate results objects used for CalculiX simulation results:
- CCX_Results
- Pipeline_CCX_Results

which is very confusing for new users. 

The first object:
- shows the results only when activated
- always shows contours on the mesh
- can't show stress and strain components
- doesn't show the color legend even though the contour plot is used (which is even more confusing and IMO rather pointless)
- can show min/max for the selected variables and uses various units which can be annoying (e.g. depending on the value, stress is shown in kPa, MPa, GPa, ...)
- supports user-defined output specified via equations which is nice but, again, there's no legend so only min/max values make sense
- can show the deformed mesh (there's a slider to change the deformation scale factor)
- shows the eigenvalues (frequencies/buckling factors) for modes from frequency and buckling analysis

The second object:
- shows the results all the time
- supports different display modes (mesh/no mesh and so on)
- can show stress and strain components
- shows the color legend
- can be significantly expanded when ParaView-based filters are added to it (including Warp to show the deformed shape)
- doesn't show the eigenvalues

I think that at some point it would be necessary to unify those results objects. Since we need the possibility to use filters, basically the results pipeline would have to be expanded to include missing options that the CCX_Results object has. Fortunately, some could be handled by new ParaView filters (like Calculator for user-defined output). We could also add min/max labels to the legend. When it comes to eigenvalue display, it would be best if it was just shown somewhere in the viewport, next to the legend.

### Full version info

```shell
OS: Windows 11 build 22621
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.34387 (Git)
Build type: Release
Branch: master
Hash: d95b26dfb5dd23eaee004b4af3cf5cf4766e7f9c
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: Polish/Poland (pl_PL)
```

### Subproject(s) affected?

FEM

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by FEA-eng** [comment]
@maxwxyz Can you remove the Bug label? This one is a feature request.

**Comment by maxwxyz** [comment]
edited all your mentions, thanks for pointing it out.

**Comment by luzpaz** [comment]
@FEA-eng is this still relevant ?

**Comment by FEA-eng** [comment]
> [@FEA-eng](https://github.com/FEA-eng) is this still relevant ?

Yes, the old Results objects have not been removed yet. Only the refactored ccx solver doesn’t use them, but it’s optional. But the new pipeline objects still need some improvements (such as animations) to support everything that the old objects include.

**Comment by maxwxyz** [comment]
@FEA-eng still relevant? I think default has the old one turned off?

**Comment by FEA-eng** [comment]
> [@FEA-eng](https://github.com/FEA-eng) still relevant? I think default has the old one turned off?

Still relevant until the old results objects are removed completely (but the new ones don't cover all their features yet). Currently, the default setting is to generate only pipelines, but the legacy objects can still be generated and are needed in some cases (e.g., for animations or when solving 1D fluid networks).
